documents/Final Report/evaluation.tex

```diff
@@ -3,20 +3,24 @@ \chapter{Evaluation and failure analysis}
-The agent can perform almost all the actions that we wanted to implement at the start of this project.
-We left out the ability to give money to another stakeholder since we noticed that realistically nobody would just give someone else money.
+The agent can perform almost all the actions that we wanted to implement at the start of this project and work as we expected.
+This is mostly because the actions are pretty straight forward like buying ground or responding to requests made by other stakeholders.
+The only thing that we did not plan was the shape in which the agent builds the new buildings. We planned on just building in the shape of the area that the agent demolished, but in th end the buildings have a custom size which is decided by the environment. This actually looks a lot better than what we had in mind so this is a good thing.
+We left out the ability to give money to another stakeholder since we noticed that realistically nobody would just give someone else money. 
+This did not have any negative impacts on the behavior of the agent, as no other stakeholder has the option to ask for money without anything in return.
+On top of that, not implementing an action has no impact anyway because the only thing that this means is that the agent is simply not able to do this action and is never called because of this. In the end we are pretty satisfied with the final result as the agent is able to do everything that we planned for it. 
 
 \section{Failure Analysis}
 Failures in the agent are usually the result of a failure in the environment, since the environment is what actually interprets the agent's actions.
 These failures more often than not throw an error which can be read in the agent's console output.
 Many possible failures are caught in the tests of the connector, but some of the failures only show up once an agent is run using that version of the connector.
 These failures most often occur when calling an action with arguments as these can be confusing and are quite prone to errors.
-Failures in the design of a percept can be quite hard to catch as it can be difficult to debug the agent and check if the percept is correct, since a percept can be quite long.
-In goal failures can be a bit harder to find since it can be difficult to debug.
-Apart from that figuring out what caused an error in the code can be quite hard to find since the eclipse plugin doesn't highlight this very well.
+Failures in the design of a percept can be quite hard to catch as it can be difficult to debug the agent and check if the percept is correct. This is because some percepts can be quite long.
+In GOAL failures can be a bit harder to find since it can be difficult to debug, because it is hard to find the failure as there are multiple factors that could have caused it. These range from goals of the agent not handled well to percepts that were different than stated in their implementation.
```

Comment:
In GOAL failures can be a bit harder to find since it can be difficult to debug, because it is hard to find the failure as there are multiple factors that could have caused it
->
Debugging can be hard in GOAL since there are multiple factors that could have caused a failure.